### PR TITLE
AJ-2002 Render local-dev config with gsm instead of vault

### PR DESCRIPTION
**Pull request by calypsomatic**
Ticket: https://broadworkbench.atlassian.net/browse/AJ-2002
<Put notes here to help reviewer understand this PR>

---
This PR updates the `render` script used for setting up config for local development to use gsm instead of vault.  Rather than relying on an external repo (first, `firecloud-develop`, later, `firecloud-develop-shim`), everything is done locally in one script.

For testing, I recommend checking out the repo in a separate directory from any rawls you may have previously checked out, to simulate starting development anew.
Then, from the root of the repo:
1. Make sure you are on Broad internet or VPN.
2. Run `./local-dev/bin/render`
3. Run `./config/docker-rysnc-local-rawls.sh`
4. Once everything seems to have built, navigate to `https://local.dsde-dev.broadinstitute.org:20443/`.  You should see the swagger page and be able to authenticate and call the APIs.

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

### CODE REVIEW

File: local-dev/templates/rawls.conf
Comment (calypsomatic) [inline local-dev/templates/rawls.conf:49]: I think we don't need anything here for local development, see https://broadinstitute.slack.com/archives/C03BCARP494/p1727097190516919?thread_ts=1727096967.429529&cid=C03BCARP494

File: local-dev/templates/sqlproxy.env
Comment (calypsomatic) [inline local-dev/templates/sqlproxy.env:1]: The name of the google project was part of the secret before, but I think it's ok for this to not be secret

File: local-dev/templates/rawls.conf
Comment (calypsomatic) [inline local-dev/templates/rawls.conf:2]: These are all just to prevent typing the same thing over and over again - and presumably to make it easy to replace if needed for some reason - but I figured it wasn't necessary and just put the values in directly

File: local-dev/bin/render
Comment (calypsomatic) [inline local-dev/bin/render]: I pulled these out early on in the theory that they might want to be replaced with a different value, but I've come to think there's no need for these to be variables and replacement, so maybe I can remove this part of the script and just hard code the values in the conf file.  Thoughts?

File: local-dev/bin/render
Comment (davidangb) [inline local-dev/bin/render]: might wanna create a helper function for `gcloud secrets versions access latest --project broad-dsde-dev --secret %1` to avoid all the copy/paste … this would also make a central place to change `broad-dsde-dev` in case anyone ever wanted to read secrets from elsewhere

**Review by davidangb — COMMENTED** [review]
I like the approach overall!

This removes the .ctmpl extension from the config files (yay!), but keeps the consul/go template syntax for variables, e.g. `{{ $rawlsOauthCredential }}`. I'm wondering if we should use a go-template CLI tool to do the variable substitution instead of doing it ourselves with `sed` … what do you think?

File: core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceAdminService.scala
Comment (calypsomatic) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceAdminService.scala:9]: These scala files were failing the `scalafmt` check after I pulled in the latest changes, so I ran `scalafmt`, but otherwise have not touched them.  Not sure how they ended up on main in the incorrect format!

File: local-dev/bin/render
Comment (davidangb) [inline local-dev/bin/render]: nit: can this use `fetch_secret`?

File: local-dev/bin/render
Comment (davidangb) [inline local-dev/bin/render]: nit: remove commented-out line

**Review by davidangb — COMMENTED** [review]
this is awesome, and unblocks development, thank you! A few things I noticed:
* If the `config` directory is missing entirely, such as on a clean checkout of the rawls repo, the script fails
* The script required VPN for me. Is that expected, perhaps due to the kubectl stuff? If so, probably worth adding a code comment or an `echo` statement to that effect
* my preference would be to have a few `echo` status lines through the script, so a user knows what it's doing and could help debug if it fails partway through
* I bet this script will fail on Windows :) I'm ok with that

File: local-dev/templates/docker-rsync-local-rawls.sh
Comment (calypsomatic) [inline local-dev/templates/docker-rsync-local-rawls.sh:129]: I've replaced `local.broadinstitute.org` with `local.dsde-dev.broadinstitute.org` everywhere in the code - please tell me if any of these need to stay `local.broadinstitute.org`!

File: local-dev/templates/docker-rsync-local-rawls.sh
Comment (calypsomatic) [inline local-dev/templates/docker-rsync-local-rawls.sh:158]: rawls-proxy was failing quietly and I didn't notice - I've added this in so it wouldn't continue to run if the proxy failed, but haven't been able to reproduce a failure to see if it even works.  Can remove, or welcome comments on a better check

Comment (davidangb) [reply local-dev/templates/docker-rsync-local-rawls.sh:158]: I'm not opposed, but all of the containers (rawls-proxy, rawls-sbt, sqlproxy, rawls-rsync-container) are necessary. I think we should check all of them or none of them to be consistent.

File: local-dev/bin/render
Comment (davidangb) [inline local-dev/bin/render]: add a `mkdir -p config` here, to ensure the `config` directory exists before we try to write to it?

**Review by davidangb — COMMENTED** [review]
looking good, two comments inline!

Before/just after this merges, make sure to announce to anyone doing rawls development about the move from `local.broadinstitute.org` to `local.dsde-dev.broadinstitute.org`

File: local-dev/templates/docker-rsync-local-rawls.sh
Comment (calypsomatic) [reply local-dev/templates/docker-rsync-local-rawls.sh:158]: Ok I added one for sqlproxy - I think for the last one you'll see in the output any failures

**Review by davidangb — APPROVED** [review]
it's all working for me, and I can auth successfully to swagger-ui. Thank you!

**Comment by ungwudik** [comment]
Looks like rawls repo is not configured to require a minimum of 2 reviews/approvals. Making a note of this for now.

**Pull request merged**
